### ISSUE TITLE
Reference tables and vocabulary server

### ISSUE THREAD

**Issue by gmaze** [euroargodev/argopy#203]
With work coming in for Argo-BGC variables, this should become important:
- Should argopy provides an easy access to Argo reference tables ?

Reference tables could be very useful to populate attributes and meta-data for raw variables retrieved from webAPI.
At this point, we hard coded what we needed. But with the huge list of variables to manage for BGC, a new internal design is required.
For instance #202 will require attributes for oyxgen.

If yes, how ? 

We could use asset files packaged with the library and updated on each release using the [NERC vocabulary server](https://vocab.nerc.ac.uk/)

**Comment by gmaze** [comment]
I'm a little bit reluctant to start from scratch an internal machinery to retrieve all Argo reference table.

@matdon17 @kamwal may be you already have some python codes to get this started ?

**Comment by gmaze** [comment]
It is not very difficult to build up a dataframe with a reference table, it can go like this:

```python
from argopy.stores import httpstore
nvs = 'https://vocab.nerc.ac.uk/collection'
fmt = '?_profile=nvs&_mediatype=application/ld+json'
get_url = "{}/{}/current/{}".format

js = httpstore().open_json(get_url(nvs, 'R01', fmt)) # Retrieve data as ld+json

# then parse:
content = {'altLabel':[], 'prefLabel': [], 'definition': [], 'deprecated': [], 'id': []}
for k in js['@graph']:
    if k['@type'] == 'skos:Collection':
        print(k['alternative'],":")
    elif k['@type'] == 'skos:Concept':
        content['altLabel'].append(k['altLabel'])
        content['prefLabel'].append(k['prefLabel']['@value']) 
        content['definition'].append(k['definition']['@value']) 
        content['deprecated'].append(k['deprecated']) 
        content['id'].append(k['@id']) 
pd.DataFrame.from_dict(content)    
```
![Screenshot 2022-04-05 at 15 16 11](https://user-images.githubusercontent.com/1956032/161762286-1bc8187a-464d-4b42-bd66-424f9bcedf0b.png)

**Comment by gmaze** [comment]
I assembled a little utility class to retrieve informations from the NVS server
It's in [argopy.utilities.ArgoNVSReferenceTables
](https://github.com/euroargodev/argopy/blob/ba173092d6c1950b32f130580a43b218ba4ff91c/argopy/utilities.py#L1857)

```python
from argopy.utilities import ArgoNVSReferenceTables

R = ArgoNVSReferenceTables()
```

Can fetch one table:
```python
R.tbl(1)  # Or R.tble('R01')
```
![Screenshot 2022-04-11 at 10 09 26](https://user-images.githubusercontent.com/1956032/162693068-b697a857-dee8-4f78-b517-9d9d1d528908.png)

or retrieve all table content:
```python
R.all_tbl()
```

But also information about a table:
```python
R.tbl_name(12)
```
or all tables name:
```python
R.all_tbl_name()
```
The list of all valid Reference tables is in:
```python
R.valid_ref
```

**Comment by gmaze** [comment]
Closed by cc8fdbe132874b71b35203053626cc29ae7d19c4
